crates/ide-completion/src/tests/type_pos.rs

```diff
@@ -719,3 +719,154 @@ pub struct S;
+        r#"
+struct Foo;
+const X: usize = 0;
+fn foo<T, const N: usize>() {}
+fn main() {
+    foo::<F$0, _>();
+}
+        "#,
+        // Method
+        r#"
+const X: usize = 0;
+struct Foo;
+impl Foo { fn bar<const N: usize, T>(self) {} }
+fn main() {
+    Foo.bar::<_, $0>();
+}
+        "#,
+    ];
+
+    for case in type_completion_cases {
+        check(
+            case,
+            expect![[r#"
+                en Enum
+                ma makro!(…) macro_rules! makro
+                md module
+                st Foo
+                st Record
+                st Tuple
+                st Unit
+                tt Trait
+                un Union
+                bt u32
+                kw crate::
+                kw self::
+            "#]],
+        )
+    }
+
+    check(
+        r#"
+const X: usize = 0;
+struct Foo;
+trait Bar {
+    type Baz<T, const X: usize>;
+}
+fn foo(_: impl Bar<Baz<F$0, 0> = ()>) {}
+        "#,
+        expect![[r#"
+            en Enum
+            ma makro!(…) macro_rules! makro
+            md module
+            st Foo
+            st Record
+            st Tuple
+            st Unit
+            tt Bar
+            tt Trait
+            un Union
+            bt u32
+            kw crate::
+            kw self::
+        "#]],
+    );
+
+    let const_completion_cases = [
+        // Function params
+        r#"
+struct Foo;
+const X: usize = 0;
+fn foo<T, const N: usize>() {}
+fn main() {
+    foo::<_, $0>();
+}
+        "#,
+        // Method params
+        r#"
+const X: usize = 0;
+struct Foo;
+impl Foo { fn bar<const N: usize, T>(self) {} }
+fn main() {
+    Foo.bar::<X$0, _>();
+}
+        "#,
+        // Associated type params
+        r#"
+const X: usize = 0;
+struct Foo;
+trait Bar {
+    type Baz<T, const X: usize>;
+}
+fn foo<T: Bar<Baz<(), $0> = ()>>() {}
+        "#,
+        // Type params
+        r#"
+const X: usize = 0;
+struct Foo<T, const N: usize>(T);
+fn main() {
+    let _: Foo::<_, $0> = todo!();
+}
+        "#,
+        // Enum variant params
+        r#"
+const X: usize = 0;
+struct Foo<T, const N: usize>(T);
+type Bar<const X: usize, U> = Foo<U, X>;
+fn main() {
+    let _: Bar::<X$0, _> = todo!();
+}
+        "#,
+        r#"
+const X: usize = 0;
+enum Foo<T, const N: usize> { A(T), B }
+fn main() {
+    Foo::B::<(), $0>;
+}
+        "#,
+        // Trait params
+        r#"
+const X: usize = 0;
+trait Foo<T, const N: usize> {}
+impl Foo<(), $0> for () {}
+        "#,
+        // Trait alias params
+        r#"
+#![feature(trait_alias)]
+const X: usize = 0;
+trait Foo<T, const N: usize> {}
+trait Bar<const M: usize, U> = Foo<U, M>;
+fn foo<T: Bar<X$0, ()>>() {}
+        "#,
+    ];
+
+    for case in const_completion_cases {
+        check(
+            case,
+            expect![[r#"
+                ct CONST
+                ct X
+                ma makro!(…) macro_rules! makro
+                kw crate::
+                kw self::
+            "#]],
+        );
+    }
```

Comment:
Let's unroll this loop and just repeat the `expect!`s, there might be future cases where they'll diverge and then it's gonna be confusing with this code here